cosmo_tester/test_suites/bootstrap_based_tests/distributed_installation_test.py

```diff
@@ -356,3 +378,77 @@ def reverse_cluster_test(cfy, cluster_machines, logger):
     for manager in cluster_machines.instances:
         toggle_cluster_node(manager, logger, disable=False)
     set_active(cluster_machines.instances[0], cfy, logger)
+
+
+def failover_cluster(cfy, distributed_installation,
+                     distributed_ha_hello_worlds, logger):
+
+    """Test that the cluster fails over in case of a service failure
+
+    - stop nginx on leader
+    - check that a new leader is elected
+    - stop mgmtworker on that new leader, and restart nginx on the former
+    - check that the original leader was elected
+    """
+    expected_master = distributed_installation.instances[-1]
+    # stop nginx on all nodes except last - force choosing the last as the
+    # leader (because only the last one has services running)
+    for manager in distributed_installation.instances[:-1]:
+        logger.info('Simulating manager %s failure by stopping'
+                    ' nginx service', manager.ip_address)
+        with manager.ssh() as fabric:
+            fabric.run('sudo systemctl stop nginx')
+        # wait for checks to notice the service failure
+        time.sleep(20)
+        wait_leader_election(distributed_installation.instances, logger)
```

Comment:
Why not use the timeout param in _wait_cluster_status?

Reply:
Fixed